ui/ducks/send/send.js

```diff
@@ -0,0 +1,1442 @@
+      // estimates gasLimit. We must use our best guess, which is represented
+      // in the gas shared constants.
+      return GAS_LIMITS.BASE_TOKEN_ESTIMATE;
+    }
+    paramsForGasEstimate.value = '0x0';
+    // We have to generate the erc20 contract call to transfer tokens in
+    // order to get a proper estimate for gasLimit.
+    paramsForGasEstimate.data = generateTokenTransferData({
+      toAddress: to,
+      amount: value,
+      sendToken,
+    });
+    paramsForGasEstimate.to = sendToken.address;
+  } else {
+    if (!data) {
+      // eth.getCode will return the compiled smart contract code at the
+      // address if this returns 0x, 0x0 or a nullish value then the address
+      // is an externally owned account (NOT a contract account). For these
+      // types of transactions the gasLimit will always be 21,000 or 0x5208
+      const contractCode = Boolean(to) && (await global.eth.getCode(to));
+      // Geth will return '0x', and ganache-core v2.2.1 will return '0x0'
+      const contractCodeIsEmpty =
+        !contractCode || contractCode === '0x' || contractCode === '0x0';
+      if (contractCodeIsEmpty) {
+        return GAS_LIMITS.SIMPLE;
+      }
+    }
+
+    paramsForGasEstimate.data = data;
+
+    if (to) {
+      paramsForGasEstimate.to = to;
+    }
+
+    if (!value || value === '0') {
+      // ??? Assuming that the value cannot be nullish or 0 to properly
+      // estimate gasLimit?
+      paramsForGasEstimate.value = '0xff';
+    }
+  }
+
+  // If we do not yet have a gasLimit, we must call into our background
+  // process to get an estimate for gasLimit based on known parameters.
+
+  paramsForGasEstimate.gas = addHexPrefix(
+    multiplyCurrencies(blockGasLimit, 0.95, {
+      multiplicandBase: 16,
+      multiplierBase: 10,
+      roundDown: '0',
+      toNumericBase: 'hex',
+    }),
+  );
+  try {
+    // call into the background process that will simulate transaction
+    // execution on the node and return an estimate of gasLimit
+    const estimatedGasLimit = await estimateGas(paramsForGasEstimate);
+    const estimateWithBuffer = addGasBuffer(
+      estimatedGasLimit.toString(16),
+      blockGasLimit,
+      1.5,
+    );
+    return addHexPrefix(estimateWithBuffer);
+  } catch (error) {
+    const simulationFailed =
+      error.message.includes('Transaction execution error.') ||
+      error.message.includes(
+        'gas required exceeds allowance or always failing transaction',
+      );
+    if (simulationFailed) {
+      const estimateWithBuffer = addGasBuffer(
+        paramsForGasEstimate.gas,
+        blockGasLimit,
+        1.5,
+      );
+      return addHexPrefix(estimateWithBuffer);
+    }
+    log.error(error);
+    throw error;
+  }
+}
+
+export async function getERC20Balance(token, accountAddress) {
+  const contract = global.eth.contract(abi).at(token.address);
+  const usersToken = (await contract?.balanceOf(accountAddress)) ?? null;
```

Comment:
Nit: it doesn't seem like it's possible for `contact` to be falsy here :thinking: 
```suggestion
  const usersToken = (await contract.balanceOf(accountAddress)) ?? null;
```